src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -0,0 +1,278 @@
+    public virtual T M<T, U>() where T : class where U : class, T => null;
+}
+class Derived : Base
+{
+    public override U M<T, U>() => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override U M<T, U>() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_03()
+        {
+            var source = @"
+class Base<T> where T : class
+{
+    public virtual T M() => null;
+}
+class Derived<T, U> : Base<T> where T : class where U : class, T
+{
+    public override U M() => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override U M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(8, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_04()
+        {
+            var source = @"
+class N { }
+class Base
+{
+    public virtual N M() => null;
+}
+class Derived<T> : Base where T : N
+{
+    public override T M() => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (9,23): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override T M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(9, 23)
+                );
+            CreateCompilation(source, parseOptions: TestOptions.WithCovariantReturns).VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void CovariantReturns_05()
+        {
+            var source = @"
+class Base
+{
+    public virtual object M => null;
+}
+class Derived : Base
+{
+    public override string M => null;
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+                // (8,28): error CS1715: 'Derived.M': type must be 'object' to match overridden member 'Base.M'
```

Comment:
>// (8,28): error CS1715: 'Derived.M': type must be 'object' to match overridden member 'Base.M' [](start = 16, length = 95)

Are we not supporting covariant returns for properties? Otherwise why there is a difference by comparison to CovariantReturns_01 scenario?

Reply:
This actually looks like a bug since we report no error for the ```TestOptions.WithCovariantReturns```` scenario. We want to have a hint that switching to a different language version is a way out.

---
In reply to: [416167876](https://github.com/dotnet/roslyn/pull/43576#discussion_r416167876) [](ancestors = 416167876)

Reply:
This is addressed in iteration 1 of [https://github.com/dotnet/roslyn/pull/43673](https://github.com/dotnet/roslyn/pull/43673)

---
In reply to: [416169487](https://github.com/dotnet/roslyn/pull/43576#discussion_r416169487) [](ancestors = 416169487,416167876)